collector/logical_port_collector_test.go

```diff
@@ -0,0 +1,138 @@
+		logicalPort := manager.LogicalPort{
+			Id:              response.ID,
+			DisplayName:     response.DisplayName,
+			LogicalSwitchId: response.LogicalSwitchID,
+		}
+		logicalPorts = append(logicalPorts, logicalPort)
+	}
+	return manager.LogicalPortListResult{
+		Results: logicalPorts,
+	}, nil
+}
+
+func (c *mockLogicalPortClient) GetLogicalPortOperationalStatus(lportID string, localVarOptionals map[string]interface{}) (manager.LogicalPortOperationalStatus, error) {
+	for _, res := range c.responses {
+		if res.ID == lportID {
+			return manager.LogicalPortOperationalStatus{
+				Status: res.Status,
+			}, res.Error
+		}
+	}
+	return manager.LogicalPortOperationalStatus{}, errors.New("error")
+}
+
+func buildLogicalPortResponse(id string, status string, err error) mockLogicalPortResponse {
+	return mockLogicalPortResponse{
+		ID:              fmt.Sprintf("%s-%s", fakeLogicalPortID, id),
+		DisplayName:     fmt.Sprintf("%s-%s", fakeLogicalPortDisplayName, id),
+		LogicalSwitchID: fmt.Sprintf("%s-%s", faceLogicalSwitchID, id),
+		Status:          status,
+		Error:           err,
+	}
+}
+
+func TestLogicalPortCollector_GenerateLogicalPortStatusMetrics(t *testing.T) {
+	testcases := []struct {
+		logicalPortListError error
+		logicalPortResponses []mockLogicalPortResponse
+		expectedMetrics      []logicalPortStatusMetric
+	}{
+		{
+			logicalPortListError: nil,
+			logicalPortResponses: []mockLogicalPortResponse{
+				buildLogicalPortResponse("01", "UP", nil),
+				buildLogicalPortResponse("02", "DOWN", nil),
+				buildLogicalPortResponse("03", "UNKNOWN", nil),
+			},
+			expectedMetrics: []logicalPortStatusMetric{
+				{
+					ID:              "fake-logical-port-id-01",
+					Name:            "fake-logical-port-name-01",
+					LogicalSwitchID: "fake-logical-switch-id-01",
+					Status:          1.0,
+				}, {
+					ID:              "fake-logical-port-id-02",
+					Name:            "fake-logical-port-name-02",
+					LogicalSwitchID: "fake-logical-switch-id-02",
+					Status:          0.0,
+				}, {
+					ID:              "fake-logical-port-id-03",
+					Name:            "fake-logical-port-name-03",
+					LogicalSwitchID: "fake-logical-switch-id-03",
+					Status:          0.0,
+				},
+			},
+		}, {
+			logicalPortListError: nil,
+			logicalPortResponses: []mockLogicalPortResponse{
+				buildLogicalPortResponse("01", "UP", nil),
+				buildLogicalPortResponse("02", "UP", errors.New("error get logical port status")),
+			},
+			expectedMetrics: []logicalPortStatusMetric{
+				{
+					ID:              "fake-logical-port-id-01",
+					Name:            "fake-logical-port-name-01",
+					LogicalSwitchID: "fake-logical-switch-id-01",
+					Status:          1.0,
+				},
+			},
+		}, {
+			logicalPortListError: errors.New("error list logical ports"),
+			logicalPortResponses: []mockLogicalPortResponse{
+				buildLogicalPortResponse("01", "UP", nil),
+				buildLogicalPortResponse("02", "UP", errors.New("error get logical port status")),
```

Comment:
shouldn't we just need to pass `nil` here, since we want to validate error list logical ports on positive lportResponse, unless we also want to validate list error on negative lportResponse

in the later case, we also have to change the case here: https://github.com/jk8s/nsxt_exporter/pull/34/files#diff-c55f744c88d64079d05e209ad2c62212R133

Reply:
yes it should be nil, forgot to remove the error